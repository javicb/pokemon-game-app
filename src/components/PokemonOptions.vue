<template>
  <div class="container">
    <ul>
      <li>1</li>
      <li>2</li>
      <li>3</li>
      <li>4</li>
    </ul>
  </div>
</template>

<script>
export default {

}
</script>

<style scoped>
ul {
  list-style-type: none;
  padding: 0px;
}
li {
    background-color: white;
    border-radius: 6px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
    padding: 1%;

}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.container {
    display: flex;
    justify-content: center;
}


</style>