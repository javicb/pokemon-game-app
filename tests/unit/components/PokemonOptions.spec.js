import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions';
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons
      }
    });
  });

  test('should be match with snapshot', () => {

    // Assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should display all four options correctly', () => {

    // Act
    const liTags = wrapper.findAll('li');

    // Assert
    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe('Bulbasaur');
    expect(liTags[1].text()).toBe('Ivysaur');
    expect(liTags[2].text()).toBe('Venusaur');
    expect(liTags[3].text()).toBe('Charmander');
  });


  test('should emit "selection" with its payload', () => {

    // Arrange
    const [li1, li2, li3, li4] = wrapper.findAll('li');

    // Act
    li1.trigger('click');
    li2.trigger('click');
    li3.trigger('click');
    li4.trigger('click');
    const emitted = wrapper.emitted('selection');

    // Assert
    expect(emitted).toHaveLength(4);
    expect(emitted[0]).toEqual([1]);
    expect(emitted[1]).toEqual([2]);
    expect(emitted[2]).toEqual([3]);
    expect(emitted[3]).toEqual([4]);
  });
})
