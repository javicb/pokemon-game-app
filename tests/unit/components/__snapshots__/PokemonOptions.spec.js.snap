// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component should be match with snapshot 1`] = `
<div class="options-container">
  <ul>
    <li class="toCapitalFirst">bulbasaur</li>
    <li class="toCapitalFirst">ivysaur</li>
    <li class="toCapitalFirst">venusaur</li>
    <li class="toCapitalFirst">charmander</li>
  </ul>
</div>
`;
